errors/error.go

```diff
@@ -43,3 +45,41 @@ type HTTPError interface {
 	Text() string
 	Code() int
 }
+
+type HTTPClientError struct {
+	code    int
+	text    string
+	rawResp string
+}
+
+func NewHTTPClientError(code int, body []byte) error {
+	err := HTTPClientError{
+		code:    code,
+		rawResp: string(body),
+		// it can be overwritten
+		text: "an unexpected error occurred, please check logs for details",
+	}
+	type ErrResp struct {
```

Comment:
```suggestion
	type ErrorResponse struct {
```

Also, could you move this type out of the function?